src/main/java/org/klodnicki/service/BookService.java

```diff
@@ -1,20 +1,38 @@
 package org.klodnicki.service;
 
-import org.klodnicki.entity.Book;
+import org.klodnicki.entity.BookInfo;
 import org.klodnicki.repository.BookRepository;
 
+import java.util.List;
+
 public class BookService {
 
     private final BookRepository bookRepository = new BookRepository();
 
-    public void add(String title, String author, String isbn, String publisher, int publicationYear, int edition,
+    public void add(String title, String author, String isbn, String publisher, int publicationYear, String edition,
                     String genre, String description, String language, int copiesNumber) {
 
         if(title.equals("") || author.equals("")) {
             throw new IllegalArgumentException("Title and author must have a value.");
         }
 
-        bookRepository.add(new Book(title, author, isbn, publisher, publicationYear, edition, genre, description,
+        bookRepository.add(new BookInfo(title, author, isbn, publisher, publicationYear, edition, genre, description,
                 language, copiesNumber));
     }
+
+    public BookInfo findBookByTitleAndAuthor(String title, String author) {
+        return bookRepository.findBookByTitleAndAuthor(title, author).orElseThrow();
```

Comment:
You can't use `orElseThrow` like that. We talked about it on last lesson and also in previous projects. Reach to me if you need additional help.